packages/headless/src/features/search/search-actions.ts

```diff
@@ -98,19 +98,19 @@ export const executeSearch = createAsyncThunk<
   addEntryInActionsHistory(state);
 
   const {analyticsClientMiddleware, preprocessRequest, logger} = config.extra;
-  const {description: eventDescription, action: analyticsAction} =
-    await searchAction.prepare({
-      getState: () => config.getState(),
-      analyticsClientMiddleware,
-      preprocessRequest,
-      logger,
-    });
+  const {description: eventDescription} = await searchAction.prepare({
+    getState: () => config.getState(),
+    analyticsClientMiddleware,
+    preprocessRequest,
+    logger,
+  });
```

Comment:
Ok, now I remember the reason why I added the whole "prepare" mechanism. The description here is meant to contain the same custom data as the analytics event we send, so I wanted to make sure they're consistent with eachother.

In `coveo.analytics`, I manually ensured that *some* parameter outside of customData was using the latest state, but it was the exception.

Instead, we should probably make it so that *everything* outside of `customData` uses the latest state.

With the current changes, the custom data may be different in the `/v2` call's `analytics.customData` object, and in the `/search` call's `customData` object.

Reply:
Alternatively, it may be worth revisiting whether we really want to include analytics in `/v2` calls

Reply:
Discussed in scrum: What's important is the action cause. Custom data that changes as a result of the query are obviously not something that can be reported to the API that we use to perform a request.